staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go

```diff
@@ -97,6 +98,13 @@ func (a statusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Obj
 
 	v := obj.GetObjectKind().GroupVersionKind().Version
 
+	// ratcheting validation of x-kubernetes-list-type value map and set
+	if newErrs := structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], uNew.Object); len(newErrs) > 0 {
+		if oldErrs := structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], uOld.Object); len(oldErrs) == 0 {
```

Comment:
I'm not even 100% sure this is reachable, but the `uOld = nil // as a safety precaution, continue with validation if uOld self cannot be cast` path above means `uOld` could possibly be nil here, and this would panic

if `uOld` is nil, I think we should enforce `newErrs`

Reply:
```
	var oldObject map[string]interface{}
	if !ok {
		oldObject = nil
	} else {
		oldObject = uOld.Object
	}
```
Is this better?